### DISCUSSION TITLE
Can I get the fee structure for wdpost penalty fee

### DISCUSSION THREAD

**Discussion by jzmexec** [filecoin-project/community#339] [Developers]
Can I get the fee structure for wdpost penalty fee? 
Can I get a way to calculate storage provider's potential penalty?
Is it in source code? I'm a begainer, isn't find in source code.

**Comment by rjan90** [comment]
Hey @jzmexec :smile: 

You can check the [specs document](https://spec.filecoin.io/#section-systems.filecoin_mining.sector.sector-faults), there are also some links to the source code in there. In short it is:

**Sector fault fee:** This fee is paid per sector per day while the sector is in a faulty state. This fee is not paid the first day the system detects the fault allowing a one day grace period for recovery without fee. The size of the sector fault fee is slightly more than the amount the sector is expected to earn per day in block rewards. If a sector remains faulty for more than 14 consecutive days, the sector will pay a termination fee and be removed from the chain state. As storage miner reliability increases above a reasonable threshold, the risk posed by these fees decreases rapidly.

**Comment by jzmexec** [comment]
@rjan90 Thanks, I got it. But can I get the detailed algorithm for wdpost penalty fee in every day. Like this:
![image](https://user-images.githubusercontent.com/71239926/137304426-399808bb-f3b8-43d8-aa09-46c66e389c7e.png)
